src/taskgraph/decision.py

```diff
@@ -227,7 +227,9 @@ def get_decision_parameters(graph_config, options):
     # ..but can be overridden by the commit message: if it contains the special
     # string "DONTBUILD" and this is an on-push decision task, then use the
     # special 'nothing' target task method.
-    if "DONTBUILD" in commit_message and options["tasks_for"] == "hg-push":
+    if "DONTBUILD" in commit_message and (
+        options["tasks_for"] == "hg-push" or options["tasks_for"] == "github-push"
+    ):
```

Comment:
Nit: Python provides the `in` keyword[1] to make this condition less verbose. Here's an example[2] Let's use it here too 🙂 

[1] https://docs.python.org/3/reference/expressions.html#in
[2] https://github.com/taskcluster/taskgraph/blob/034305f5f77d2c6a0c2d0ca851ae683a9cb07d01/src/taskgraph/docker.py#L49

Reply:
Thanks for the kind words!
I hope the test is correct.
I'm also not sure if I have to include the `github-pull-request` to the `DONTBUILD` list in `decision.py` or not.

Reply:
The test looks great! We could split the cases into multiple ones but we would have to convert the test class into a `pytest` case. No need to do it in this PR, this could be done in a follow-up one 🙂 

>  I'm also not sure whether I have to include the github-pull-request to the DONTBUILD list in decision.py or not.

That's a great question! Most of the usage of our `DONTBUILD` usage comes from CD (continuous deployment). For example: this task[1] created that `DONTBUILD` commit. We also use `DONTBUILD` for commits related to documentation and trivial fixes.

We could allow `DONTBUILD` on pull requests but I'm unsure of the side effects. A PR could contain multiple commits. If the last one has `DONTBUILD`, then the whole stack of commits won't be tested at all. I'm not sure we should do it today. That said, we may want to make taskgraph smarter about pull requests and detect whether a PR contains multiple commits and still build if one commit omits `DONTBUILD`. However, this improvement is definitely outside of the scope of https://github.com/taskcluster/taskgraph/issues/189 🙂

[1] https://firefox-ci-tc.services.mozilla.com/tasks/NU4C9bT8TLm6e-re1YfSdA
[2] https://treeherder.mozilla.org/jobs?repo=mozilla-beta&revision=c708a6fd4bfbf7dccf2a70b52fbbf2dfb0056a02

Reply:
IMO DONTBUILD on PRs should be ignored.

Reply:
I see. Thanks for the thorough explanation.